### ISSUE TITLE
Hosting Rewrites to not-us-central1 should print a warning

### ISSUE THREAD

**Issue by igorakkerman** [firebase/firebase-tools#1158]
I tried setting up a simple rewrite to a simple function (hello-world grade). The function was successfully deployed in europe-west1. The rewrite is triggered but redirects to a Google authentication page. Here's the message from the function's log: 

    Billing account not configured. External network is not accessible and quotas are severely limited. Configure billing account to remove these restrictions

If I deploy the function in the default region (us-central1), the redirect works just fine.

Why is a billing account required? 
How would such a call be billed?
Is it possible to stay in europe-west1 for both, the static part and the function?

**Comment by bkendall** [comment]
Thanks for the feedback. I've updated the bug for tracking purposes. I think all the information required to do this is actually available in the code, it's just a matter of implementing the warning. I can't say when/if we'd get around to this, but it seems like a reasonable idea.

**Comment by compojoom** [comment]
Instead of adding a warning, the firebase team should concentrate on offering different regions for hosting. The warning won't fix the core issue.
 
Most of our customers are in europe. We need to have rewrites, because we have an API endpoint and I don't want to explain why the url is europe-west1-project-id.cloudfunctions.net/ instead of just api.mydomain.com

and the other more important negative side effect is that our data is in EU, it has to travel all the way to US and then back to EU...

**Comment by thechenky** [comment]
Hi @compojoom thanks for the feedback. I understand how this must be frustrating for you. I'll pass your request off to @Memeriaj who can talk more about whether adding more regions for hosting rewrites is on the roadmap.

**Comment by Memeriaj** [comment]
We're thinking about it, like Michael said.  If it we're just enabling region support then it wouldn't be that hard, just give us another string that is what region you want it in.  But unfortunately there is more that has to happen before that becomes feasible.

Right now all of Firebase Hosting's origin servers are located in `us-central` (for legacy reasons that I'd rather not get into), but this is something that we're currently working on.  So any function that Hosting rewrites to has to go from where ever the request was made, hit the closest CDN PoP, then get redirected to `us-central` to hit the Hosting origin, make a request to the Function endpoint, then make its way back up the stack.  Having the Function located anywhere besides `us-central` will only incur additional latency when it travels from `us-central` and back.

Once we have the Hosting origin located in other places that will be one of the things we work on to enable.

**Comment by fazpu** [comment]
@Memeriaj - Please could you give us any estimates when Hosting origin will be located outside of US?

The whole Firebase environment is amazing, but this make us think twice before switching to it. Thank you

**Comment by bkendall** [comment]
It's a general rule that we cannot comment on timelines, but we are still actively working on this. Thanks for your patience.

(Also, it doesn't seem all that polite to post your question in multiple issues; once is enough. Thanks.)

**Comment by joehan** [comment]
Hosting rewrites support all regions today: https://firebase.google.com/docs/hosting/full-config#rewrite-functions
